pkg/libvmi/vm.go

```diff
@@ -74,3 +74,9 @@ func WithDataVolumeTemplate(datavolume *v1beta1.DataVolume) VMOption {
 		)
 	}
 }
+
+func WithVMNamespace(namespace string) VMOption {
```

Comment:
```suggestion
func WithNamespace(namespace string) VMOption {
```

Reply:
Is this method necessary? AFAIU, the namespace should be deducted from the VMI passed to generate the VM

Reply:
> I find this to hint of a problem.
> Why is this needed? (considering Alice prev comment)

So the VMI is fixed in this factory, can't pass a namespace through it.
This is what allows to control the namespace of the resulting VM
(Which I assume we all is a necessary customizable)

Reply:
Usually on creation client command, there is a need to specify the namespace.
It should be rare to require setting the manifest with the namespace.

Reply:
> I find this to hint of a problem.

Yep.. I have some ideas, for now, removing this and continue to chase this down